### ISSUE TITLE
Transport of Cofactors Across Peroxisomal Membrane

### ISSUE THREAD

**Issue by Devlin-Moyer** [SysBioChalmers/Human-GEM#640]
## Background:

Human-GEM currently has the following cytosol <-> peroxisome transport reactions for CoA, FAD, FADH<sub>2</sub>, NAD<sup>+</sup>, NADH, AMP, ADP, and ATP:

ID | Reaction | Genes
---|---|---
`MAR04915` | CoA [c] <=> CoA [x] | none
`MAR07790` | FAD [c] <=> FAD [x] | none
`MAR07788` | FADH2 [c] <=> FADH2 [x] | none
`MAR07771` | NAD+ [x] --> NAD+ [c] | none
`MAR07775` | NADH [c] --> NADH [x] | none
`MAR07783` | AMP [c] <=> AMP [x] | none
`MAR07782` | ADP [c] <=> ADP [x] | none
`MAR07785` | ATP [c] <=> ATP [x] | none
`MAR03473` | AMP [x] + ATP [c] --> AMP [c] + ATP [x] | `ENSG00000100372`
`MAR04908` | ADP [c] + ATP [x] <=> ADP [x] + ATP [c] | `ENSG00000103024 or ENSG00000103202 or ENSG00000143156 or ENSG00000155085 or ENSG00000172113 or ENSG00000239672 or ENSG00000243678`

## Problems:

According to [this paper](https://doi.org/10.1042/BJ20111420), SLC25A17 (`ENSG00000100372`) can exchange most pairs of these metabolites between the cytosol and peroxisome, with the notable exceptions of ATP, NADH, and (indirectly) FADH<sub>2</sub>. While they didn't directly test transport of FADH<sub>2</sub>, they did directly demonstrate that SLC25A17 can transport NAD<sup>+</sup> but not NADH (so `MAR07775` should be removed), so it seems reasonable to infer that SLC25A17 is similarly capable of transporting FAD but not FADH<sub>2</sub> (so `MAR07788` should also be removed). SLC25A17 exchanges ADP and AMP rather than ATP and AMP, so `MAR03473` should be edited accordingly. Transport of ATP across the peroxisomal membrane is accurately accounted for by `MAR04908`, which involves different genes that are known to encode transporters that can transport ATP across the peroxisomal membrane (see the references associated with `MAR04908`), so `MAR07785` is a less accurate/complete duplicate of `MAR04908` and should be removed.

That paper also mentioned that SLC25A17 is only capable of _exchanging_ metabolites between the cytosol and peroxisome; while it can exchange cytosolic AMP or ADP for peroxisomal AMP or ADP, it cannot catalyze _net_ transport of AMP or ADP in either direction without doing antiport with a different compound. So `MAR04915`, `MAR07790`, `MAR07771`, `MAR07783`, and `MAR07782` should either be edited to be antiport with one of the other substrates of SLC25A17 or removed. I think it might be simpler/easier to follow if we just removed all of those and added a new block of antiport reactions that all had consecutive IDs and associated each of those new reactions with the old IDs of all of these reactions in the reactions.tsv file, but let me know if there's a general rule about trying to edit rather than replace existing reactions.

SLC25A17 can also exchange most of these metabolites for FMN, but no peroxisomal FMN metabolite currently exists. The [SLC25A17 paper](https://doi.org/10.1042/BJ20111420) mentioned that FMN can be produced in the peroxisome by hydrolysis of FAD by NUDT12 (`ENSG00000112874`; already in Human-GEM), and cited [this paper](https://doi.org/10.1042/bj20030441) to support that claim. Adding a peroxisomal FMN metabolite would also be useful if we follow up on the suggestion in #609 to separate all reactions catalyzed by FMN-dependent reactions into two half-reactions: one where the substrate is oxidized and FMN is reduced, and another where FMNH<sub>2</sub> is oxidized and something else is reduced (for the human peroxisomal FMN-dependent enzymes [HAO1](https://www.uniprot.org/uniprotkb/Q9UJM8/entry) and [HAO2](https://www.uniprot.org/uniprotkb/Q9NYQ3/entry), that something else would be O<sub>2</sub>).

The [SLC25A17 paper](https://doi.org/10.1042/BJ20111420) also mentions that SLC25A17 can transport PAP (adenosine 3',5'-bisphosphate), and that it could be produced by the action of NDUT7 and NDUT19 on CoA and acyl-CoAs, and no peroxisomal PAP metabolite currently exists. We could add a peroxisomal version of PAP, the CoA -> PAP reaction, and appropriate SLC25A17-mediated transport reactions, but I'm not sure that it'd be worth the effort, since plenty of reactions can produce and consume PAP in [c], none can do so in [x], and most of the other substrates of SLC25A17 are already accounted for, so not including PAP is unlikely to significantly limit their ability to move across the peroxisomal membrane. Furthermore, hydrolysis of CoA to PAP also produces phosphopantetheine, which also does not currently exist in [x], and it's not clear if that could also be transported across the peroxisomal membrane or if it would need to be degraded further before being able to leave the peroxisome.

## Proposed Changes:

- [x] Remove `MAR07785` for being a less accurate/complete duplicate of `MAR04908`
- [x] Remove `MAR07775` because there's no evidence that NADH is transported across the peroxisomal membrane
- [x] Remove `MAR07788` because there’s no evidence that FADH<sub>2</sub> is transported across the peroxisomal membrane
- [x] Remove `MAR04915`, `MAR07771`, `MAR07782`, `MAR07783`, and `MAR07790` because the only known peroxisomal transporters of those metabolites catalyze antiport, not uniport
- [x] Edit `MAR03473` to be ADP [c] + AMP [x] <-> ADP [x] + AMP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction AMP [c] <-> AMP [x], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction ADP [c] <-> ADP [x], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new FMN [x] metabolite `MAM01828x`
- [x] Add new reaction FAD [x] + H<sub>2</sub>O [x] -> FMN [x] + AMP [x], GPR: `ENSG00000112874`, reference: `PMID:22185573`
- [x] Add new reaction FMN [c] + CoA [x] <-> FMN [x] + CoA [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FMN [c] + FAD [x] <-> FMN [x] + FAD [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FMN [c] + NAD<sup>+</sup> [x] <-> FMN [x] + NAD<sup>+</sup> [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FMN [c] + AMP [x] <-> FMN [x] + AMP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FMN [c] + ADP [x] <-> FMN [x] + ADP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FAD [c] + CoA [x] <-> FAD [x] + CoA [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FAD [c] + NAD<sup>+</sup> [x] <-> FAD [x] + NAD<sup>+</sup> [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FAD [c] + AMP [x] <-> FAD [x] + AMP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction FAD [c] + ADP [x] <-> FAD [x] + ADP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction NAD<sup>+</sup> [c] + CoA [x] <-> NAD<sup>+</sup> [x] + CoA [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction NAD<sup>+</sup> [c] + AMP [x] <-> NAD<sup>+</sup> [x] + AMP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction NAD<sup>+</sup> [c] + ADP [x] <-> NAD<sup>+</sup> [x] + ADP [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction AMP [c] + CoA [x] <-> AMP [x] + CoA [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`
- [x] Add new reaction ADP [c] + CoA [x] <-> ADP [x] + CoA [c], GPR: `ENSG00000100372`, reference: `PMID:22185573`

**Comment by feiranl** [comment]
Nice! 
As for the reactions to remove:
- [ ] I suggest we keep `MAR07775`  and `MAR07788` for now, since there is not a evidence for saying it doesn't exist, the evidence now just say that NADH and FADH2 cannot be transported by SLC25A17
- [ ] Remove MAR04915, MAR07790, MAR07771, MAR07783, and MAR07782 should be only done before test all tasks can be filled.
I suggest the rule for model curation is to only make changes when they are supported by evidence and are deemed correct.

**Comment by Devlin-Moyer** [comment]
It's difficult to prove that a reaction _doesn't_ happen, and there are clearly a lot of reactions that were added to Human-GEM with no references or genes associated with them, plausibly during gap-filling at some point, so I feel like "only remove reactions that we can prove do not occur" is a questionable curation rule.

Also, consider these bits of [this paper](https://doi.org/10.3389/fcell.2015.00083)

>beta-oxidation in peroxisomes can only continue if the NADH formed in peroxisomes is reoxidized to NAD+. Ideally, a carrier system in the peroxisomal membrane catalyzing the exchange between NADH in the peroxisome and NAD+ in the cytosol would serve this purpose. However, since such a system appears to be lacking at least in higher eukaryotes, the existence of metabolite-based redox shuttles has been proposed analogous to the well-known malate/aspartate shuttle in mitochondria [...] In higher eukaryotes, however, including humans, the identity of the peroxisomal NAD(H)-redox shuttle has not been resolved definitively although evidence in favor of the existence of a lactate/pyruvate-based redox shuttle has been provided by [Baumgart et al. (1996)](https://www.frontiersin.org/articles/10.3389/fcell.2015.00083/full#B7), at least in rat liver peroxisomes (see Figure [5B](https://www.frontiersin.org/articles/10.3389/fcell.2015.00083/full#F5)). Whatever the precise identity of the peroxisomal NAD(H)-redox shuttle, ultimate reoxidation of peroxisomal NADH back into NAD+, can only be achieved in mitochondria.

**Comment by Devlin-Moyer** [comment]
The experiments done in this paper showed that, under various conditions, human peroxisomes were either more reducing or more oxidizing environments than the cytosols of the surrounding cells, which would not be possible if there was free exchange of both oxidized and reduced NAD(H), NADP(H), and/or FAD(H2) https://doi.org/10.1091/mbc.e10-11-0919

**Comment by feiranl** [comment]
Seems OK to me. Just need to make sure to check the metabolic tasks can be fulfilled. 
The task list can be found [here](https://github.com/SysBioChalmers/Human-GEM/tree/main/data/metabolicTasks), and @haowang-bioinfo  maybe you can guide how to test those tasks. I only find the RAVEN function  `checkTasks`, not sure wether this is correct.

```
taskStruct = parseTaskList('../data/metabolicTasks_Essential.txt');
[taskReport, essentialRxnMat, ~, essentialFluxes] = checkTasks(bModel,[],true,false,true,taskStruct);
```

**Comment by haowang-bioinfo** [comment]
@Devlin-Moyer it's very good to have a lot of references provided

@feiranl agree to make sure all checks should be passed, because these are essential cofactors

**Comment by Devlin-Moyer** [comment]
I can make a PR with these changes to see if they interfere with the metabolic tasks, but I'm waiting on #670 to be resolved first. #670 adds new reactions, and some of the proposed changes here add new reactions, and I want to avoid a confusing merge conflict where different new reactions share the same ID

**Comment by haowang-bioinfo** [comment]
fixed in #701
